### PR TITLE
Check total env vars before creating .env file

### PR DESCRIPTION
**Pull request by MonicaisHer**
resolves https://github.com/canonical/edgex-snap-hooks/issues/63

```go
replace github.com/canonical/edgex-snap-hooks/v2 => github.com/MonicaisHer/edgex-snap-hooks/v2 v2.2.0-beta.0.20220707123953-815231d5ada7
```

### CODE REVIEW

**Review by farshidtz — CHANGES_REQUESTED** [review]
I think you need to rebase.
```
# github.com/canonical/edgex-snap-hooks/v2 [github.com/canonical/edgex-snap-hooks/v2/options.test]
./utils.go:71:6: CopyFile redeclared in this block
        /snap/edgex-snap-hooks/x5/copy.go:28:41: previous declaration
./utils.go:89:6: CopyDir redeclared in this block
        /snap/edgex-snap-hooks/x5/copy.go:46:46: previous declaration
./utils.go:132:6: CopyFileReplace redeclared in this block
        /snap/edgex-snap-hooks/x5/copy.go:89:76: previous declaration
FAIL    github.com/canonical/edgex-snap-hooks/v2/options [build failed]
```

File: options/options_test.go [hunk 1]
```diff
@@ -242,7 +243,28 @@ func TestProcessAppConfig(t *testing.T) {
...
+	_, err := os.Stat(file)
+	if err == nil {
+		return true, nil
+	}
+	if errors.Is(err, os.ErrNotExist) {
+		// file does not exist
+		return false, nil
+	}
+	return false, err
+}
```
Comment (farshidtz) [inline options/options_test.go:256]: Why not remove this function and change the other function to:
```go
func isInFile(file string, line string) error {
	// read the whole file at once
	content := readFile(file)
	if content != "" && strings.Contains(content, line) {
		return nil
	} else {
		return fmt.Errorf("Line %s not found in %s", line, file)
	}
}
```

File: options/env.go [hunk 2]
```diff
@@ -76,6 +76,13 @@ func (cp *configProcessor) writeEnvFiles() error {
 		var buffer bytes.Buffer
 		filename := cp.filename(app)
 
+		// do not create a .env file if there are no snap options set for the app
+		// remove .env file if exists
+		if len(envVars) == 0 {
+			os.RemoveAll(filename)
```
Comment (farshidtz) [inline options/env.go:82]: Please add error handling.

**Review by farshidtz — CHANGES_REQUESTED** [review]

**Comment by farshidtz** [comment]
Updated the tests to fail when env file isn't removed:
```
$ sudo edgex-snap-hooks.test ./options
Running tests from /snap/edgex-snap-hooks/x5/snap/local/bin/test.sh
Change directory to /snap/edgex-snap-hooks/x5
--- FAIL: TestProcessAppConfig (1.86s)
    --- FAIL: TestProcessAppConfig/global_options (0.37s)
        --- FAIL: TestProcessAppConfig/global_options/set+unset (0.25s)
            options_test.go:94: 
                        Error Trace:    /snap/edgex-snap-hooks/x5/options/options_test.go:94
                                                                /snap/edgex-snap-hooks/x5/options/testing.go:912
                                                                /snap/edgex-snap-hooks/x5/options/testing.go:1049
                                                                /snap/edgex-snap-hooks/x5/options/testing.go:1253
                        Error:          Should be false
                        Test:           TestProcessAppConfig/global_options/set+unset
                        Messages:       Env file should not exist.
    --- FAIL: TestProcessAppConfig/single_app_options (0.26s)
        --- FAIL: TestProcessAppConfig/single_app_options/set+unset (0.21s)
            options_test.go:137: 
                        Error Trace:    /snap/edgex-snap-hooks/x5/options/options_test.go:137
                                                                /snap/edgex-snap-hooks/x5/options/testing.go:912
                                                                /snap/edgex-snap-hooks/x5/options/testing.go:1049
                                                                /snap/edgex-snap-hooks/x5/options/testing.go:1253
                        Error:          Should be false
                        Test:           TestProcessAppConfig/single_app_options/set+unset
                        Messages:       Env file should not exist.
FAIL
coverage: 43.0% of statements
FAIL    github.com/canonical/edgex-snap-hooks/v2/options        1.863s
FAIL
```

**Comment by MonicaisHer** [comment]
@farshidtz Thanks for your commits. Looks good.

**Review by farshidtz — APPROVED** [review]

**Pull request merged**